src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -88,6 +101,26 @@ public boolean publish(String message, String color) {
         return result;
     }
 
+    private String getTokenToUse() {
+        if (credentialsAuthTokenId != null && !credentialsAuthTokenId.isEmpty()) {
+            StringCredentials credentials = lookupCredentials(credentialsAuthTokenId);
+            if (credentials != null) {
+                logger.info("Using secret for Slack auth token.");
+                return credentials.getSecret().getPlainText();
+            }
+        }
+
+        logger.info("Using plain text for Slack auth token.");
+
+        return token;
+    }
+
+    private StringCredentials lookupCredentials(String credentialId) {
```

Comment:
Unfortunately, I don't have the required parameters to send into that function.  The way I'm doing it here seems to be a popular way to handle looking up credentials from looking at other GitHub projects.